core/src/main/java/io/opencensus/tags/Tags.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.internal.Provider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/** Class for accessing the default {@link TagsComponent}. */
+public final class Tags {
+  private static final Logger logger = Logger.getLogger(Tags.class.getName());
+
+  private static final TagsComponent tagsComponent =
+      loadTagsComponent(Provider.getCorrectClassLoader(TagsComponent.class));
+
+  private Tags() {}
+
+  public static TagContextFactory getTagContextFactory() {
+    return tagsComponent.getTagContextFactory();
+  }
+
+  // Any provider that may be used for TagsComponent can be added here.
+  @VisibleForTesting
+  @Nullable
+  static TagsComponent loadTagsComponent(ClassLoader classLoader) {
```

Comment:
I tried refactoring this method, but it didn't work very well.  I think that the most repetitive part is handling the ClassNotFoundException that can be thrown by `Class.forName`.  However, `Class.forName` needs to be called with a literal String so that it can be modified by shading tools, so it can't be moved to a helper method.  I don't think that there is enough remaining code to split out.